src/Symfony/Component/Amqp/DsnParser.php

```
@@ -0,0 +1,33 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Amqp;
+
+/**
+ * @internal
+ *
+ * @author Grégoire Pineau <lyrixx@lyrixx.info>
+ */
+final class DsnParser
+{
+    public static function parseDsn(string $dns): array
+    {
+        $parts = parse_url($dns);
```

You need to throw an exception when the result is `false`.